### ISSUE TITLE
Cannot run windows store applications

### ISSUE THREAD

**Issue by DionHo** [artoolkit/arunity5#16]
Unity Version: 5.2.0f3
ARToolkit Version: 5.3r1

Reproduction of the Problem:
  - Use MultimarkerScene from the Example Scenes
  - Build Settings: Windows Store, SDK 8.1, Development build

Host Machine Info:
  - Windows 8.1 Home, 64bit

Error Messages:
The whole output is covers about 1000 lines, Here the most important ones (IMHO)

```
...
ARController: Starting AR.
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARController: Render device: Direct3D 11.0 [level 11.0], using Unity texturing.
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARController: Starting ARToolKit video with vconf '-device=WinMC -format=BGRA -position=rear'.
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARController (native): ARController::startRunning(): called, start running
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARController (native): VideoSource::configure(): video Source video configuration: "-device=WinMC -format=BGRA -position=rear"
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARController (native): VideoSource::configure(): video Source camera parameters buffer: 176 bytes
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


Using supplied video config "-device=WinMC -format=BGRA -position=rear".

 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


Requesting images in BGRA format.

 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


'AdventuAR.exe' (Win32): Loaded 'C:\Windows\SysWOW64\Windows.Media.dll'. Cannot find or open the PDB file.
'AdventuAR.exe' (Win32): Loaded 'C:\Windows\SysWOW64\DevDispItemProvider.dll'. Cannot find or open the PDB file.
'AdventuAR.exe' (Win32): Loaded 'C:\Windows\SysWOW64\DDORes.dll'. Cannot find or open the PDB file.
'AdventuAR.exe' (Win32): Loaded 'C:\Windows\SysWOW64\DefaultDeviceManager.dll'. Cannot find or open the PDB file.
ARWrap::ARVideo::WindowsMediaCapture::initDevices(): found 1 video capture devices
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARWrap::ARVideo::WindowsMediaCapture::StartCapture(): error-no video devices available, exiting returning false
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


Error starting capture.

 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARWrap::ARVideo::WindowsMediaCapture::~dtor(): called
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARController (native): [error]ARWrap::ARToolKitVideoSource::open(): arVideoOpen unable to open connection to camera using configuration '-device=WinMC -format=BGRA -position=rear', exiting returning false
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARController (native): [error]ARController::startRunning(): Error: unable to open video source, exiting, returning false
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


ARController: Error starting running
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)


AROrigin: Found 1 markers eligible to become base marker.
 
(Filename: C:/buildslave/unity/build/artifacts/generated/Metro/runtime/UnityEngineDebugBindings.gen.cpp Line: 70)
...
```

I hope this can be of help. If you have problems reproducing this error, please tell me ;)

**Comment by BoiseBaked** [comment]
DionHo, tell me what the PC hardware you using, the PC running "Windows 8.1 Home, 64bit." Does the PC have two cameras? If not, does the PC has one webcam? IF so, the webcam is probably not designated as a "rear" camera. If the PC is a desktop variety, it may not have any webcam whatsoever.

Also, tell me about the nature of the project you're building and trying to execute. What kind of Unity platform are you building to, i.e. Build Settings: Platform, Type, SDK, "Build and Run on".

**Comment by wallywyyoung** [comment]
@DionHo Are you still around?

**Comment by DionHo** [comment]
Hi, sorry for not replying.

I tried to reproduce this problem a while ago, but there were other problems coming up. I also wanted to test, if I still have errors when I use the recently released version 5.3.1. 

To the Hardware: The PC has one webcam, which is most probably not designated as rear-camera. All I tried was to build the MultimarkerScene from the examples delivered with arunity. The Problem only occurs when I choose the Windows Store - Platform (8.1, Development build, Create C# Project).

I will try to build the new FistiCuffs-Example the Windows Store - Platform on my PC in a few days and then reply again.

Thank you for replying to my issue :)

**Comment by wallywyyoung** [comment]
@BoiseBaked seeing if you got the update. =)

**Comment by BoiseBaked** [comment]
The only thing I would add is that the app has to enumerate all camera devices available on the platform where the app is deployed. And the app must allocate an existing device by ID. All the sample apps do not do this for all platforms, so the source has to be updated to do it. Also, your creating a Windows Desktop app, use the PC Standalone platform in the Unity Build Settings. If you targeting Windows 8.1 Windows Store or Windows 10 Windows Store, select the appropriate platform, SDK and Type from the Unity build settings. Stay away from targeting Windows 8.0. In general, if you're targeting Windows Store, use Windows Store Universal.

**Comment by mFontoura** [comment]
@BoiseBaked  I also can't get an app running correctly on WinPhone 8.1.

I'm using a Lumia 1520 that has a 1080p video camera. However when I open my Unity App, the camera is smaller than the screen at 320x240.

Also, dispite my NFT markers working on the Unity Player they don't work in the Lumia. Maybe because the camera doesn't even auto focus?

All this is after I export the Unity Project to Visual Studio 2015.

Despite ARController having "videoConfigurationWindowsStore0" set to "-device=WinMC -format=BGRA -position=rear -width=1920 -height=1080", the log shows that when the app runs this variable is changed to "-device=WinMC -format=BGRA -position=rear". I think this happens on ARToolKit?

Anyway, in "start()" of the ARController I set this variable back has I wanted. This way the camera is in fullscreen and the video in 1080p, however is extremely slow... 1fps or less.

Hope this helps. Any question please ask, I'm urgently trying to solve this.

**Comment by BoiseBaked** [comment]
First things first. Please list your Unity Pro version, all, the example ARTK app your working with, if you are working with one, the exact Unity editor Build Settings and the scene your building against.
Also, are you saying that none of this occurred when you used Visual Studio 2013?

Right now, I'm behind the curve for Unity for Windows. I need to get a Unity 5 Pro license for my Windows 10 devbox and I need to upgrade from Visual Studio 2013 to 2015. I'm working on it.

**Comment by mFontoura** [comment]
Sorry forgot to mention the versions. Right now I'm using Unity Editor 5.3.0f4, you don't need a Unity 5 license, since Unity5 all pro features and included in the free version. I didn't tried it in VS2013, only 2015 Community.

**Comment by mFontoura** [comment]
Ok, no idea what is happening. 

On a different computer, I installed the same version of unity, and VS2015 with winphone 8.1 SDK. 
- before my first export form unity I set parameters to "-device=WinMC -format=BGRA -position=rear -width=1920 -height=1080".
- On VS the build went with no errors and on the phone the camera was full screen 1080p!! :+1:  However, again, the FPS was super low and it didn't recognize the NFT tracker.

Btw, the app only has a 3D cube on top of the tracker.

Correct me if I'm wrong but, on ARController.cs the variable "videoConfigurationWindowsStore0" is the default in case nothing is set on the player? In that case part of the problem is that I can't change the values of the device after the first export.
